vmm/src/lib.rs

```diff
@@ -1202,12 +1243,58 @@ impl Vmm {
     }
 }
 
+/* Create ptys here and add the slave paths to Serial, Console and DebugConsole
+    configs. This allows apply_landlok to add the slave paths to landlock rules
+    later.
+*/
+fn create_ptys(config: Arc<Mutex<VmConfig>>) -> result::Result<(), LandlockError> {
+    if config.lock().unwrap().console.mode == ConsoleOutputMode::Pty {
```

Comment:
You can do `let config = config.lock().unwrap()` once at the start of the function.